### ISSUE TITLE
Option to skip formatting CSS

### ISSUE THREAD

**Issue by helgenlechner** [jhildenbiddle/css-vars-ponyfill#160]
Thank you for this great library! We've been using it for a couple years now. I've been doing some performance debugging recently and noticed that quite a lot of time is spent on "sanitizing" the CSS. `parseCss` alone takes around 2.5 seconds in our case. Since we always roll out minified CSS, all the cleaning and trimming in the function `selector`, for example, doesn't result in a change. I've tried simply skipping that step and was able to reduce the time needed by `parseCss` by over 50%.

I believe that minifying CSS is quite a common use case these days. What do you think about adding an option to the ponyfill that would skip all these sanitizing steps? Maybe it could even be recommended as a best practice in the ReadMe, since this would improve runtime performance considerably.

I'm happy to work on this, since the improvements would be worthwhile for our projects. Do you perhaps have a sense of what parts of the code could be skipped with this option?

**Comment by jhildenbiddle** [comment]
Thank you for the kind words, @helgenlechner. Much appreciated, and happy to hear the ponyfill has worked well for you.

The behavior in the [`selector()`](https://github.com/jhildenbiddle/css-vars-ponyfill/blob/master/src/parse-css.js#L107-L128) function you are referring to is used to detect individual selectors within in a comma-separated list of selectors. Sanitized CSS is a side effect, not the goal. There are definitely optimizations that can be done though.

First, I've copied the `selector()` source below and added comments to explain each mutation:

```js
// Match selector
const m = match(/^(("(?:\\"|[^"])*"|'(?:\\'|[^'])*'|[^{])+)/);

if (m) {
    return m[0]
        .trim()
        // Remove comments
        .replace(/\/\*([^*]|[\r\n]|(\*+([^*/]|[\r\n])))*\*\/+/g, '')
        // Replace comma in comma-separated lists with marker
        .replace(/"(?:\\"|[^"])*"|'(?:\\'|[^'])*'/g, function(m) {
            return m.replace(/,/g, '\u200C');
        })
        // Create array from comma-separated list of selectors
        .split(/\s*(?![^(]*\)),\s*/)
        // Restore comma in comma-separated lists
        .map(function(s) {
            return s.replace(/\u200C/g, ',');
        });
}
```

Here are a few CSS selector examples that would trip up the parser without the code above:

```css
html,
/* :root, */
body {
 ...
}

.foo,
.bar, /* Comment, with a comma */
.baz {
  ...
}

body[attr="foo,bar"] { ... }
```

One issue with `selector()` code above is that it processes all selectors the same way. We can optimize the process by doing simple checks first to determine if we need to do more complex mutations. For example, if a simple test for `/*` within the selector string fails, we don't need execute a complex regular expression to replace all comments in a selector. The end result is that complex selectors may take bit longer, but simple selectors will be processed much faster. I've implemented these changes in [branch 160](https://github.com/jhildenbiddle/css-vars-ponyfill/tree/160):

https://github.com/jhildenbiddle/css-vars-ponyfill/blob/b16b784d07fb9c8a32d65a62664f8efcaaa16f26/src/parse-css.js#L107-L157

Feel free to check out the branch, run `npm build`, and test the results of the new build on your project. For reference, I ran a few rudimentary tests locally and saw a 50% increase in `selector()` performance.
